### PR TITLE
JBPM-5862: Stunner - Diagram tree explorer improvements.

### PR DESCRIPTION
**Pull request by alepintus**
This is a code refactor due to a uberfire tree improvement. Should be merged with this [PR of uberfire](https://github.com/AppFormer/uberfire/pull/708) ; as they have compilation dependencies.
For more information see [this ticket](https://issues.jboss.org/browse/JBPM-5862).

Thanks.

### CODE REVIEW

**Comment by romartin** [comment]
ok to test

**Comment by romartin** [comment]
Jenkins please retest this

File: guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/advnavigator/TreeNavigator.java [hunk 1]
```diff
@@ -111,38 +111,38 @@ public void callback( final NavigatorContent response ) {
...
-            newDirectory.addItem( TreeItem.Type.LOADING, LAZY_LOAD );
+            final FSTreeItem newDirectory = parent.addItem( FSTreeItem.FSType.FOLDER, child.getFileName() );
+            newDirectory.addItem( FSTreeItem.FSType.LOADING, LAZY_LOAD );
             newDirectory.setUserObject( child );
         }
 
         public void addFile( final Path child ) {
             checkCleanupLoading();
 
             //Util.getHeaderSafeHtml( images.file(), child.getFileName() )
```
Comment (ederign) [inline guvnor-structure/guvnor-structure-client/src/main/java/org/guvnor/structure/client/advnavigator/TreeNavigator.java]: @alepintus @romartin when you guys have a chance, delete dead comments like these.

File: guvnor-structure/guvnor-structure-client/src/test/java/org/guvnor/structure/client/editors/fileexplorer/FileExplorerViewImplTest.java [hunk 2]
```diff
@@ -38,116 +39,125 @@
...
+            public FSTreeItem getChild(int i) {
+                return cti.get(i);
             }
 
-            @Override
-            protected TreeItem makeChild( final Type type,
-                                          final String value ) {
-                return new TreeItem( type,
-                                     value ) {
+            /*
```
Comment (ederign) [inline guvnor-structure/guvnor-structure-client/src/test/java/org/guvnor/structure/client/editors/fileexplorer/FileExplorerViewImplTest.java]: dead code.

File: guvnor-structure/guvnor-structure-client/src/test/java/org/guvnor/structure/client/editors/fileexplorer/FileExplorerViewImplTest.java [hunk 3]
```diff
@@ -38,116 +39,125 @@
...
+                        @Override
+                        public Iterable<TreeItem> getChildren() {
+                            return cti;
+                        }
+            */
+            //@Override
```
Comment (ederign) [inline guvnor-structure/guvnor-structure-client/src/test/java/org/guvnor/structure/client/editors/fileexplorer/FileExplorerViewImplTest.java]: dead code.

**Comment by romartin** [comment]
Oh sorry @ederign , I did not realize about that dead code, you're right. Please @alepintus , can you please tomorrow provide a new PR just for fixing the above comments? Thanks guys!!

**Comment by alepintus** [comment]
@ederign @romartin sure! I'll make a new one asap.

**Pull request merged**